ddtrace/contrib/botocore/patch.py

```diff
@@ -0,0 +1,65 @@
+"""
+Trace queries to aws api done via botocore client
+"""
+
+# project
+import ddtrace
+
+# 3p
+import wrapt
+import botocore.client
+import pkg_resources
+from distutils.version import LooseVersion
+
+from ...ext import AppTypes
+from ...ext import http
+
+# Original botocore client class
+_Botocore_client = botocore.client.BaseClient
+
+DEFAULT_SERVICE = "aws"
+SPAN_TYPE = "botocore"
+
+
+def patch():
+    # Checking for the version compatibility before patching
+    if LooseVersion(pkg_resources.get_distribution("botocore").version) >= LooseVersion("1.4.51"):
```

Comment:
i don't think we want version matching code at this layer. instead let the patching function fail if a patchable isn't available - and `patch_all` will respond to this gracefully